File: newcoder/question/NC41.cpp
class Solution {
public:
    /**
     *
     * @param arr int整型vector the array
     * @return int整型
     */
    int maxLength(vector<int>& arr) {
        // write code here
        map<int, int> mp;
        int left = 0, right = 1, maxLen = 1;
        while (right < arr.size() && left <= right) {
            if (mp.count(arr[right])) {
                while (arr[left] != arr[right]) {
                    mp.erase(arr[left]);
                    ++left;
                }
                ++left;
            }
            mp[arr[right]] = 1;
            maxLen = max(maxLen, right - left + 1);
            ++right;
        }
        return maxLen;
    }
};

File: 双指针/nowcoder/NC 41.js
function maxLength( arr ) {
    // write code here
    let left = 0, max = 0, map = new Map();
    for (let right = 0; right < arr.length; right++) {
        if (map.has(arr[right]) && map.get(arr[right]) >= left) {
            left = map.get(arr[right]) + 1;
        }
        map.set(arr[right], right);
        max = Math.max(max, right - left + 1);
    }
    return max;
}
module.exports = {
    maxLength : maxLength
};